compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1995,11 +1995,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             .as_local()
             .and_then(|def_id| hir.maybe_body_owned_by(def_id))
             .map(|body_id| hir.body(body_id));
-        let is_async = self
-            .tcx
-            .generator_kind(generator_did)
-            .map(|generator_kind| matches!(generator_kind, hir::GeneratorKind::Async(..)))
-            .unwrap_or(false);
+        let is_async = self.tcx.async_kind(generator_did).is_some();
```

Comment:
should/could this not be moved down to where it is used; either the call to, or inside of `note_obligation_cause_for_async_await` ? as it doesn't seem to be otherwise used